pyproject.toml

```diff
@@ -59,14 +59,53 @@ script-files = ["scripts/download_gleam.py", "scripts/make_flat_spectrum_eor.py"
 
 [tool.setuptools_scm]
 
-[tool.isort]
-profile = "black"
-
 [darglint]
 docstring_style = "numpy"
 
-[tool.black]
-skip_magic_trailing_comma = true
-
 [tool.pytest.ini_options]
 addopts = "--ignore=scripts"
+
+[tool.ruff.lint]
+select = [
+    "E", # pycodestyle
+    "W", # pycodestyle warnings
+    "F", # Pyflakes
+    "D", # pydocstyle
+    "UP", # pyupgrade
+    "B", # flake8-bugbear
+    "A", # flake8-builtins
+    "C4", # flake8-comprehensions
+    "N", # pep8-naming
+    "SIM", # flake8-simplify
+    "I", # isort
+    # "C90", # McCabe complexity. Consider for the future
+]
+ignore = [
+    "N806", # non-lowercase variable (we use N* for axes lengths)
+    "B028", # no-explicit-stacklevel for warnings
+    "SIM108", # prefer ternary opperators. I find them difficult to read.
+    "D203", # one-blank-line-before-class. we use two.
+    "D212", # multi-line-summary-first-line. We put it on the second line.
+]
+
+[tool.ruff.lint.per-file-ignores]
+"tests/*" = ["D"] # Don't require docstrings for tests
+"docs/*.py" = ["D", "A"] # Don't require docstrings or worry about builtins for docs
+"setup.py" = ["D"] # Don't require docstrings for setup.py
+
+[tool.ruff.format]
+skip-magic-trailing-comma = true
+
+[tool.ruff.lint.pycodestyle]
+max-line-length = 88
```

Comment:
is 88 what we had before? feels small. Ah yeah it was in the .flake8. Probably just getting confused because pyuvsim is 100. Is there any real reason they should differ? I feel like having a common style across the rasg repos would be nice.

Reply:
It turns out that pyuvsim is the outlier. pyuvdata is 88

Reply:
would you look at that. Well i'd like to formally casts my vote to bump the other two to 100